server/src/definition/SourceDefinitionProvider.ts

```diff
@@ -62,15 +63,30 @@ export class SourceDefinitionProvider implements DbtNodeDefinitionProvider {
   getTableDefinitions(source: string, table: string, dbtSources: ManifestSource[], tableSelectionRange: Range): DefinitionLink[] | undefined {
     const foundSource = dbtSources.find(s => s.sourceName === source && s.name === table);
     if (foundSource) {
+      const sourceDefinitionFileContent = fs.readFileSync(foundSource.originalFilePath, 'utf8');
+      const sourceDefinitionFileLines = sourceDefinitionFileContent.split('\n');
+      const targetRange = this.getSourceRange(sourceDefinitionFileLines, table);
       return [
         LocationLink.create(
           path.join(foundSource.rootPath, foundSource.originalFilePath),
-          DbtDefinitionProvider.MAX_RANGE,
-          DbtDefinitionProvider.MAX_RANGE,
+          targetRange ?? DbtDefinitionProvider.MAX_RANGE,
+          targetRange ?? DbtDefinitionProvider.MAX_RANGE,
           tableSelectionRange,
         ),
       ];
     }
     return undefined;
   }
+
+  private getSourceRange(lines: string[], tableName: string): Range | undefined {
```

Comment:
Can we add some unit tests for this method? 

Isn't this process slow when we match each line with regex?

Reply:
Added unit test.

Reply:
Now inclusion of a string will be checked before regex match.